### DISCUSSION TITLE
hi Raghu, I am getting below error while executing shell script to launch vms.

### DISCUSSION THREAD

**Discussion by mareedu868** [learndevopsonline/learndevopsonline#56] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/108469186/219369553-2eec3b1e-6e1f-4016-a9c2-0bc61048c89f.png)

[ centos@workstation ~/roboshop-shell ]$ bash create-ec2-servers.sh
sed: can't read route53.json: No such file or directory
Server Created - FAILED - DNS RECORD - catalogue-dev.mydevopslearning.online

18.207.235.231 | 172.31.56.202 | t2.micro | https://github.com/mareedu868/roboshop-shell.git
[ centos@workstation ~/roboshop-shell ]$ cat create-ec2-servers.sh
#!/bin/bash

##### Change these values ###
ZONE_ID="Z0439315OZEA1H4ZPOVL"
DOMAIN="mydevopslearning.online"
SG_NAME="allow-all"
env=dev
#############################

create_ec2() {
  PRIVATE_IP=$(aws ec2 run-instances \
      --image-id ${AMI_ID} \
      --instance-type t3.micro \
      --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=${COMPONENT}}, {Key=Monitor,Value=yes}]" "ResourceType=spot-instances-request,Tags=[{Key=Name,Value=${COMPONENT}}]"  \
      --instance-market-options "MarketType=spot,SpotOptions={SpotInstanceType=persistent,InstanceInterruptionBehavior=stop}"\
      --security-group-ids ${SGID} \
      | jq '.Instances[].PrivateIpAddress' | sed -e 's/"//g')

  sed -e "s/IPADDRESS/${PRIVATE_IP}/" -e "s/COMPONENT/${COMPONENT}/" -e "s/DOMAIN/${DOMAIN}/" route53.json >/tmp/record.json
  aws route53 change-resource-record-sets --hosted-zone-id ${ZONE_ID} --change-batch file:///tmp/record.json 2>/dev/null
  if [ $? -eq 0 ]; then
    echo "Server Created - SUCCESS - DNS RECORD - ${COMPONENT}.${DOMAIN}"
  else
     echo "Server Created - FAILED - DNS RECORD - ${COMPONENT}.${DOMAIN}"
     exit 1
  fi
}

**Comment by mareedu868** [comment]
I have removed env and tried but the same error. Here is my updated script.

[ centos@workstation ~/roboshop-shell ]$ cat create-ec2-servers.sh
#!/bin/bash

##### Change these values ###
ZONE_ID="Z0439315OZEA1H4ZPOVL"
DOMAIN="mydevopslearning.online"
SG_NAME="allow-all"
#############################



create_ec2() {
  PRIVATE_IP=$(aws ec2 run-instances \
      --image-id ${AMI_ID} \
      --instance-type t3.micro \
      --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=${COMPONENT}}, {Key=Monitor,Value=yes}]" "ResourceType=spot-instances-request,Tags=[{Key=Name,Value=${COMPONENT}}]"  \
      --instance-market-options "MarketType=spot,SpotOptions={SpotInstanceType=persistent,InstanceInterruptionBehavior=stop}"\
      --security-group-ids ${SGID} \
      | jq '.Instances[].PrivateIpAddress' | sed -e 's/"//g')

  sed -e "s/IPADDRESS/${PRIVATE_IP}/" -e "s/COMPONENT/${COMPONENT}/" -e "s/DOMAIN/${DOMAIN}/" route53.json >/tmp/record.json
  aws route53 change-resource-record-sets --hosted-zone-id ${ZONE_ID} --change-batch file:///tmp/record.json 2>/dev/null
  if [ $? -eq 0 ]; then
    echo "Server Created - SUCCESS - DNS RECORD - ${COMPONENT}.${DOMAIN}"
  else
     echo "Server Created - FAILED - DNS RECORD - ${COMPONENT}.${DOMAIN}"
     exit 1
  fi
}


## Main Program
AMI_ID=$(aws ec2 describe-images --filters "Name=name,Values=Centos-8-DevOps-Practice" | jq '.Images[].ImageId' | sed -e 's/"//g')
if [ -z "${AMI_ID}" ]; then
  echo "AMI_ID not found"
  exit 1
fi

SGID=$(aws ec2 describe-security-groups --filters Name=group-name,Values=${SG_NAME} | jq  '.SecurityGroups[].GroupId' | sed -e 's/"//g')
if [ -z "${SGID}" ]; then
  echo "Given Security Group does not exit"
  exit 1
fi


for component in catalogue cart user shipping payment frontend mongodb mysql rabbitmq redis dispatch; do
  COMPONENT="${component}"
  create_ec2
done

**Reply by LokeshViswa** [reply]
while executing Instead of the cat command try with bash command. It will work

**Comment by mareedu868** [comment]
Hey Lokesh, I think you didnt look at my screenshot or the message properly..I have executed the script with bash and I just showed the content in the script using cat .

**Comment by learninguser** [comment, marked as answer]
Hi,
Your script is unable to locate route53.json file which is used for the configuration of spot instances on your AWS dashboard. On cross-verifying your github repo (roboshop-shell), I am unable to locate route53.json file in your repository.

![image](https://user-images.githubusercontent.com/38400460/219395258-b69dc6bc-8c50-4d0c-85a9-26be8a105e99.png)

So kindly place both the files i.e. create-ec2-servers.sh and route53.json in the same directory and execute `bash create-servers.sh` .

Please keep us posted with the update on whether its working or not.

Kind regards
Pavan Kumar.

**Comment by mareedu868** [comment]
Thanks Pavan.. I missed to copy the file..Its working now.
